Building with FFmpeg 7.0 fails

**Describe the problem**
FFmpeg 7.0 has been released. Building `openboard` against FFmpeg 7.0 fails with this error message: 
```
/build/openboard-git/src/OpenBoard/src/podcast/ffmpeg/UBFFmpegVideoEncoder.cpp: In member function ‘bool UBFFmpegVideoEncoder::init()’:
/build/openboard-git/src/OpenBoard/src/podcast/ffmpeg/UBFFmpegVideoEncoder.cpp:523:65: error: ‘AVCodecContext’ {aka ‘struct AVCodecContext’} has no member named ‘channels’
  523 |         mAudioOutBuffer = av_audio_fifo_alloc(c->sample_fmt, c->channels, c->frame_size);
      |                                                                 ^~~~~~~~
/build/openboard-git/src/OpenBoard/src/podcast/ffmpeg/UBFFmpegVideoEncoder.cpp: In member function ‘void UBFFmpegVideoEncoder::processAudio(QByteArray&)’:
/build/openboard-git/src/OpenBoard/src/podcast/ffmpeg/UBFFmpegVideoEncoder.cpp:643:60: error: ‘AVCodecContext’ {aka ‘struct AVCodecContext’} has no member named ‘channels’
  643 |                                              codecContext->channels, outSamplesCount,
      |                                                            ^~~~~~~~
/build/openboard-git/src/OpenBoard/src/podcast/ffmpeg/UBFFmpegVideoEncoder.cpp: In member function ‘void UBFFmpegVideoEncoder::finishEncoding()’:
/build/openboard-git/src/OpenBoard/src/podcast/ffmpeg/UBFFmpegVideoEncoder.cpp:739:18: warning: ‘int avcodec_close(AVCodecContext*)’ is deprecated [-Wdeprecated-declarations]
  739 |     avcodec_close(mVideoCodecContext);
      |     ~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~
In file included from /build/openboard-git/src/OpenBoard/src/podcast/ffmpeg/UBFFmpegVideoEncoder.h:26,
                 from /build/openboard-git/src/OpenBoard/src/podcast/ffmpeg/UBFFmpegVideoEncoder.cpp:22:
/usr/include/libavcodec/avcodec.h:2387:5: note: declared here
 2387 | int avcodec_close(AVCodecContext *avctx);
      |     ^~~~~~~~~~~~~
/build/openboard-git/src/OpenBoard/src/podcast/ffmpeg/UBFFmpegVideoEncoder.cpp:743:22: warning: ‘int avcodec_close(AVCodecContext*)’ is deprecated [-Wdeprecated-declarations]
  743 |         avcodec_close(mAudioCodecContext);
      |         ~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~
/usr/include/libavcodec/avcodec.h:2387:5: note: declared here
 2387 | int avcodec_close(AVCodecContext *avctx);
      |     ^~~~~~~~~~~~~

```

**Versions**
OS: Arch Linux
openboard: 1.7.1 and 753eadad57b92ce510b80f07775f20818d5f3201 (current `dev`)
GCC: 14.1.1+r58+gfc9fb69ad62-1
qt6-base: 6.7.1-4
ffmpeg: 2:7.0-3

Built against `extra-staging` on Arch Linux, i.e. not yet generally available (preview version).

**Build commands**
Built using the CMake community build. Full instructions: [PKGBUILD](https://aur.archlinux.org/cgit/aur.git/tree/PKGBUILD?h=openboard)

**Suggestions, solutions**
From what I can tell, it's not as big a jump as from 4.4 to 5.0 - The error above can likely be fixed by changing `codecContext->channels` to `codecContext->ch_layout->nb_channels`. There might be more errors afterwards. I'll look into it later.

--- a/src/podcast/ffmpeg/UBFFmpegVideoEncoder.cpp
+++ b/src/podcast/ffmpeg/UBFFmpegVideoEncoder.cpp
@@ -520,7 +520,12 @@ bool UBFFmpegVideoEncoder::init()
         }
 
         // Buffer for resampled/converted audio
-        mAudioOutBuffer = av_audio_fifo_alloc(c->sample_fmt, c->channels, c->frame_size);
+#if LIBAVCODEC_VERSION_INT < AV_VERSION_INT(61, 3, 100)
+        int nb_channels = c->channels;
+#else
+        int nb_channels = c->ch_layout.nb_channels;
+#endif
+        mAudioOutBuffer = av_audio_fifo_alloc(c->sample_fmt, nb_channels, c->frame_size);
     }
 
 
@@ -639,8 +644,13 @@ void UBFFmpegVideoEncoder::processAudio(QByteArray &data)
     uint8_t ** outSamples = nullptr;
     int outSamplesLineSize;
 
+#if LIBAVCODEC_VERSION_INT < AV_VERSION_INT(61, 3, 100)
+    int nb_channels = codecContext->channels;
+#else
+    int nb_channels = codecContext->ch_layout.nb_channels;
+#endif
     ret = av_samples_alloc_array_and_samples(&outSamples, &outSamplesLineSize,
-                                             codecContext->channels, outSamplesCount,
+                                             nb_channels, outSamplesCount,
                                              codecContext->sample_fmt, 0);
     if (ret < 0) {
         qWarning() << "Could not allocate audio samples" << avErrorToQString(ret);

--- a/src/podcast/ffmpeg/UBFFmpegVideoEncoder.cpp
+++ b/src/podcast/ffmpeg/UBFFmpegVideoEncoder.cpp
@@ -746,11 +746,19 @@ void UBFFmpegVideoEncoder::finishEncoding()
     av_write_trailer(mOutputFormatContext);
     avio_close(mOutputFormatContext->pb);
 
+#if LIBAVCODEC_VERSION_INT < AV_VERSION_INT(61, 3, 100)
     avcodec_close(mVideoCodecContext);
+#else
+    avcodec_free_context(&mVideoCodecContext);
+#endif
     sws_freeContext(mSwsContext);
 
     if (mShouldRecordAudio) {
+#if LIBAVCODEC_VERSION_INT < AV_VERSION_INT(61, 3, 100)
         avcodec_close(mAudioCodecContext);
+#else
+        avcodec_free_context(&mAudioCodecContext);
+#endif
         swr_free(&mSwrContext);
     }
 

